### PR TITLE
Update lifecycle and phpunit

### PR DESCRIPTION
**Pull request by nevil93**
added two migrations and lifecycle events for them, also create a Streamline service and unit test for it

### CODE REVIEW

File: app/Entities/Repositories/MessageRepository.php [hunk 1]
```diff
@@ -3,12 +3,36 @@
...
+    public function searchFiltering($searchRequest)
     {
-        return $this->_em->createQuery('SELECT m.message FROM App\Entities\Message m')
-                         ->getResult();
+        $queryBuilder = $this->_em->createQueryBuilder();
+
+
+
+        if (empty($searchRequest)) {
+            $result = $queryBuilder->select('p')
```
Comment (egorbwork) [inline app/Entities/Repositories/MessageRepository.php]: select and from a common operations. I recommend to extract them before if.

File: app/Http/Controllers/DataController.php [hunk 2]
```diff
@@ -4,26 +4,32 @@
...
 use Doctrine\ORM\EntityManager;
-use App\Entities\Person;
+//use App\Entities\Person;
 use App\Entities\Message;
+use App\Services\Streamline;
 
 class DataController extends Controller
 {
-    public function displayData(EntityManager $em)
+    protected $entityManager;
```
Comment (egorbwork) [inline app/Http/Controllers/DataController.php]: Type hints will simplify your life in future.

File: app/Http/Controllers/DataController.php [hunk 3]
```diff
@@ -4,26 +4,32 @@
...
+
+    public function __construct(EntityManager $entityManager)
     {
+        $this->entityManager = $entityManager;
+    }
+
+    public function search(Request $request, Streamline $streamline)
```
Comment (egorbwork) [inline app/Http/Controllers/DataController.php]: Why do you need `Streamline`?

File: app/Services/Streamline.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            ];
+
+        foreach ($person->getMessages() as $message) {
+            $data['messages'][] = $message->getContent();
+        }
+        return $data;
+    }
+
+
+}
```
Comment (egorbwork) [inline app/Services/Streamline.php]: We will have to configure your local IDE to ensure trailing spacde in the end.

File: app/Services/Streamline.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Services;
+
+
+use App\Entities\Person;
+
+
+class Streamline
```
Comment (egorbwork) [inline app/Services/Streamline.php]: Please revise name. Before looking inside I was not able to understand what it is doing.

File: app/Services/Streamline.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    public function convertToArray(Person $person)
+    {
+        $data = [
+            'id' => $person->getId(),
+            'name' => $person->getName(),
+            'email' => $person->getEmail()
+            ];
+
+        foreach ($person->getMessages() as $message) {
+            $data['messages'][] = $message->getContent();
```
Comment (egorbwork) [inline app/Services/Streamline.php]: Let's also keep id of the message.

File: storage/proxies/__CG__AppEntitiesPerson.php [hunk 7]
```diff
@@ -0,0 +1,279 @@
+<?php
```
Comment (egorbwork) [inline storage/proxies/__CG__AppEntitiesPerson.php]: Add this folder to gitignore and remove them. It will be a pain to have this filess in git.

File: tests/Feature/StreamlineTest.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use App\Entities\Person;
+use App\Services\Streamline;
+
+class StreamlineTest extends TestCase
+{
+    public function testIfAreEquals()
```
Comment (egorbwork) [inline tests/Feature/StreamlineTest.php]: Please investigate the php unit conventions for naming

File: tests/Feature/StreamlineTest.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        $streamline = new Streamline();
+        $person = new Person();
+
+        $arr = [
+            'id' => null,
+            'name' => null,
+            'email' => null
+        ];
+        $this->assertEquals($arr, $streamline->convertToArray($person));
```
Comment (egorbwork) [inline tests/Feature/StreamlineTest.php]: It was interesting trial. But in final you need to test it with objects which contains data not empty ones. Please create another test which instead of empty object will work with entity with data. And the result will be build from array with data you expects there.

File: tests/Feature/StreamlineTest.php [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public function testIfReturnArray()
```
Comment (egorbwork) [inline tests/Feature/StreamlineTest.php]: This and next one are not useful tests, but leave them as an experience trial.

File: app/Entities/Repositories/MessageRepository.php [hunk 11]
```diff
@@ -3,12 +3,33 @@
 namespace App\Entities\Repositories;
 
 use Doctrine\ORM\EntityRepository;
+use App\Entities\Person;
 
 class MessageRepository extends EntityRepository
```
Comment (egorbwork) [inline app/Entities/Repositories/MessageRepository.php]: You are searching the person so move it to the person repository.

File: tests/Feature/FromObjectToArrayServiceAnotherTest.php [hunk 12]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Tests\Feature;
```
Comment (egorbwork) [inline tests/Feature/FromObjectToArrayServiceAnotherTest.php]: Find best practices for the namespaces in the unit tests and apply.

File: tests/Feature/FromObjectToArrayServiceAnotherTest.php [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+use App\Services\FromObjectToArrayService;
+
+class FromObjectToArrayServiceAnotherTest extends TestCase
+{
+    public function testIsCorrectArrayWhatIsReturned()
+    {
+        $streamline = new FromObjectToArrayService();
+        $person = new Person();
+
+        $person->setName('Nicolai')->setEmail('nicolai@gmail.com');
```
Comment (egorbwork) [inline tests/Feature/FromObjectToArrayServiceAnotherTest.php]: This data is not enough to fully test your service. It is made not to test, but `na otiebisi`.

File: tests/Feature/FromObjectToArrayServiceAnotherTest.php [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+
+        $arr = [
```
Comment (egorbwork) [inline tests/Feature/FromObjectToArrayServiceAnotherTest.php]: I recommend to build it from values not from using getters.
